### DISCUSSION TITLE
overlayPopup in sidebar is moved when main content scrolled

### DISCUSSION THREAD

**Discussion by yureckey** [primefaces/community#628] [PrimeVue]
When I have sidebar with overlayPopup (or confirmPopup) if I scroll main content - popup is moved with it (obviously, it shouldn't).
I think this is a bug, but maybe at least there is an easy workaround?
sample:
https://stackblitz.com/edit/2gg6wz?file=package.json,src%2FApp.vue

**Comment by tugcekucukoglu** [comment]
For now, you can try adding appendTo=self to the OverlayPanel and importing it into the Sidebar. We will investigate what we can do on the core side.

**Reply by yureckey** [reply]
Interesting workaround! It works for default sidebar, but when I tried for sidebar with position="right" it placed overlay way off screen 😢

**Comment by yureckey** [comment]
> For now, you can try adding appendTo=self to the OverlayPanel and importing it into the Sidebar. We will investigate what we can do on the core side.

The sort-of-workaround would be if I can dynamically change blockScroll property of sidebar (I can change it now of course, but it will take effect only when sidebar is closed/opened while I have sidebar always open). I tried this
const mask = document.querySelector('.p-sidebar-mask');
if (mask) mask.classList.add('p-sidebar-mask-scrollblocker');
but it has no effect on block scroll
